lib/block-supports/layout.php

```diff
@@ -36,9 +37,20 @@ function gutenberg_register_layout_support( $block_type ) {
  *
  * @return string                                CSS style.
  */
-function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support = false, $gap_value = null, $should_skip_gap_serialization = false ) {
+function gutenberg_get_layout_style( $block_name, $selector, $layout, $has_block_gap_support = false, $gap_value = null, $should_skip_gap_serialization = false ) {
 	$layout_type = isset( $layout['type'] ) ? $layout['type'] : 'default';
 
+	// If there is no block-level value for blockGap,
+	// but a global styles value available for blockGap,
+	// use the latter.
+	if ( $has_block_gap_support && empty( $gap_value ) ) {
+		$block_global_styles = gutenberg_get_global_styles( array( 'blocks', $block_name, 'spacing' ) );
```

Comment:
Just wondering, could we move this check to happen in `gutenberg_render_layout_support_flag` instead, before we parse / sanitize the`$gap_value` variable, so that by the time we're in this function, it's already the right gap value? That we might not need to update the signature for `gutenberg_get_layout_style`.

Reply:
Good call 👍